### ISSUE TITLE
Automatically downloading full resolution images option

### ISSUE THREAD

**Issue by Ope30** [gnosygnu/xowa#164]
Is there a option in XOWA that lets you download full resolution images?

**Comment by gnosygnu** [comment]
Sort of, but it's manual, and needs a few extra steps.

When you see a thumbnail image in an article
* Click on the image. It will take you to a commons.wikimedia.org page inside XOWA
* On this commons page, click the image again. This will download the full resolution image and open it in your image editor.

I can make this "simpler", but keep in mind that full resolution images are heavy, and I don't want to do something like "download all the full resolution images on this page". Some pages can easily be a few hundred MB (for example, anything related to visual arts)

**Comment by Ope30** [comment]
> When you see a thumbnail image in an article

> Click on the image. It will take you to a commons.wikimedia.org page inside XOWA

> On this commons page, click the image again. This will download the full resolution image and open it in your image editor.

True. I just thought of a option that you can enable/disable to automatically download full resolution once you click on a image that gets you to Commons. Would be cool to see.

> I can make this "simpler", but keep in mind that full resolution images are heavy, and I don't want to do something like "download all the full resolution images on this page". Some pages can easily be a few hundred MB (for example, anything related to visual arts)

Yeah, the file size. :P Fortunately XOWA is able to delete previous images downloaded from Commons. Or you can do it manually by just deleting the "file" folder (?)!

**Comment by gnosygnu** [comment]
> True. I just thought of a option that you can enable/disable to automatically download full resolution once you click on a image that gets you to Commons. Simply just a option if you want to have it. Would be cool to see!

Okay! I put it down on my todo list, but can't make any promises of when it will happen

> Fortunately XOWA is able to delete previous images downloaded from Commons. Or you can do it manually by just deleting the "file" folder (?)!

Yeah, just to be clear, XOWA downloads the file and puts it in 2 places:
* A temporary copy somewhere in the /xowa/file directory.
     * This temporary copy is needed in order to show on the web page
     * This temporary copy will get deleted once XOWA exceeds the file cache max of 100 MB
* A "permanent" copy that gets saved to /xowa/wiki/en.wikipedia.org/en.wikipedia.org-file-user.xowa (or wherever the wiki is)
     * This "permanent" copy is used next time the page is viewed and the temporary file doesn't exist
     * This "permanent" copy only gets deleted when the file gets deleted

That said, the original point still remains. I don't like the idea of XOWA downloading hundreds of MB of images just by opening one page. So I "force" the user to download individual pages one by one by going through the steps outlined above.

**Comment by markusmaresch** [comment]
I believe we all should be reminded about WHY we love XOWA. It is not only a good replica of the online version, mostly it is for OFFLINE viewing !! (If I am online, I browse the original anyway ... )
**Offline viewing and having a reasonable compromise on speed and database size is key**. We should not forget those initial goals. An online dependency should not start to creep into in, in my mind .. my 2 cents.

**Comment by gnosygnu** [comment]
Hey Markus, thanks as always for the kind words regarding XOWA. I do agree with your sentiment -- and yes, even I use the online version. :)

I will, of course, do my best to avoid adding any online dependencies. Just to allay some concerns: If this feature is introduced, it would only be activated by a config. It would be disabled by default. I think that should help mitigate any online requirement. A user-config should also address any concerns about slowing down speed and increasing database size. (The user would have to deliberately choose to activate this config setting)

With that said, I think this option will probably not happen anytime in the near future, as I'm still dealing with other issues / bugs, while trying to carve out some time to work on XOWA.

Let me know if the above did not address your concerns, or if I missed anything.

Thanks!

**Comment by markusmaresch** [comment]
Hi - thanks for your kind reply !  No no, that's fine - we are on the same page :-)
